File: C Sharp/Blink/Blink/Core/ReceivePacket.cs
using System;

namespace Net.Qiujuer.Blink.Core
{
    /// <summary>
    /// Receive Packet
    /// </summary>
    public abstract class ReceivePacket : BlinkPacket
    {
        private readonly long mId;
        private String mHash;

        public ReceivePacket(long id, byte type, long len)
            : base(type)
        {
            mId = id;
            mLength = len;
        }

        public long GetId()
        {
            return mId;
        }

        /// <summary>
        /// Set Packet Hash Code
        /// </summary>
        /// <param name="hashCode">HashCode</param>
        public void SetHash(String hashCode)
        {
            mHash = hashCode;
        }

        /// <summary>
        /// Get Packet Hash Code
        /// </summary>
        /// <returns>HashCode</returns>
        public String GetHash()
        {
            return mHash;
        }

        /// <summary>
        /// On Receiver receive some info buffer call this
        /// </summary>
        /// <param name="buffer">Info data</param>
        /// <param name="offset">Buffer offset</param>
        /// <param name="count">Buffer Count</param>
        internal virtual void WriteInfo(byte[] buffer, int offset, int count) { }

        /// <summary>
        /// Receiver write buffer
        /// </summary>
        /// <param name="buffer">Buffer</param>
        /// <param name="offset">Buffer offset</param>
        /// <param name="count">Buffer Count</param>
        internal abstract void Write(byte[] buffer, int offset, int count);
    }
}

File: Java/Blink/Blink/src/net/qiujuer/blink/core/ReceivePacket.java
/*
 * Copyright (C) 2014 Qiujuer <qiujuer@live.cn>
 * WebSite http://www.qiujuer.net
 * Created 04/16/2015
 * Changed 04/19/2015
 * Version 1.0.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.qiujuer.blink.core;

/**
 * Receive Packet
 */
public abstract class ReceivePacket extends Packet {
    private final long mId;
    private String mHash;

    public ReceivePacket(long id, byte type, long len) {
        super(type);
        mId = id;
        mLength = len;
    }

    public long getId() {
        return mId;
    }

    /**
     * Set Packet Hash Code
     *
     * @param hashCode HashCode
     */
    public void setHash(String hashCode) {
        mHash = hashCode;
    }

    /**
     * Get Packet Hash Code
     *
     * @return HashCode
     */
    public String getHash() {
        return mHash;
    }

    /**
     * On Receiver receive same info buffer call this
     *
     * @param buffer Info data
     * @param offset Buffer offset
     * @param count  Buffer Count
     */
    public void writeInfo(byte[] buffer, int offset, int count) {
    }

    /**
     * Receiver write buffer
     *
     * @param buffer Buffer
     * @param offset Buffer offset
     * @param count  Buffer Count
     */
    public abstract void write(byte[] buffer, int offset, int count);
}
